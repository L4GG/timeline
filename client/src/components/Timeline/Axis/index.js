// @flow
import React, { type Node } from 'react';
import styled from 'styled-components';

import { color } from '../../../styles';

type Props = {|
  children?: Node,
  height: number,
|};

export const WIDTH = 20;

const Bar = styled.div`
  background: ${color.alto};
  position: relative;
  width: ${WIDTH}px;
`;

const Axis = ({ children, height }: Props) => (
  <Bar style={{ height }}>{children}</Bar>
);

export default Axis;
