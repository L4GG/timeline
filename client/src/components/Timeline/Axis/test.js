// @flow
/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Axis from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Axis height={10} />, div);
});

it('renders correctly', () => {
  const tree = renderer.create(<Axis height={10} />).toJSON();

  expect(tree).toMatchSnapshot();
});
