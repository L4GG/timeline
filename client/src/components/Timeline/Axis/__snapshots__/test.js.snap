// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa cUhefd"
  style={
    Object {
      "height": 10,
    }
  }
/>
`;
